src/golem-network/golem-network.test.ts

```diff
@@ -187,4 +187,19 @@ describe("Golem Network", () => {
       verify(mockPayment.releaseAllocation(allocation)).never();
     });
   });
+  describe("disconnect()", () => {
+    it("reuses the same promise if called multiple times", async () => {
+      const glm = getGolemNetwork();
+      const glmSpy = spy(glm);
+      when(glmSpy["startDisconnect"]()).thenResolve();
+      expect(glm["disconnectPromise"]).toBeUndefined();
+      const promise1 = glm.disconnect();
+      const promise2 = glm.disconnect();
+      const promise3 = glm.disconnect();
+      expect(glm["disconnectPromise"]).toBeDefined();
+      await Promise.all([promise1, promise2, promise3]);
```

Comment:
Parallel runs passes the test, but sequential fails:
```ts
await promise1;
await promise2;
```

Reply:
This is the correct logic in my opinion, if you await the promise then the disconnect finishes and on the next run it's the same as running `disconnect` on a newly created `GolemNetwork` (without `connect` first), nothing should happen in that case

Reply:
What I mean is I think this should be valid:
```ts
const glm = new GolemNetwork()
await glm.connect();
await glm.disconnect();
await glm.disconnect() // nothing happens
await glm.connect();
await glm.disconnect();
```

Reply:
ok, I agree, so disconnecting on an unconnected `GolemNetwork` should not result in any action on the network, so it should either be resolved immediately or give an error, e.g. `GolemUserError("Network is not connected")`

Reply:
same for rental pool
```ts
const pool = new ResourceRentalPool({ ... });
await pool.ready();
await pool.drainAndClear(); // finalizes all rentals
await pool.drainAndClear(); // nothing happens
await pool.ready();
// pool is ready to use again
await pool.drainAndClear(); // finalizes all rentals
```